Source/DafnyTestGeneration/Utils.cs

```diff
@@ -84,18 +84,18 @@ public static class Utils {
     /// <summary>
     /// Parse a string read (from a certain file) to a Dafny Program
     /// </summary>
-    public static async Task<Program> /*?*/ Parse(ErrorReporter reporter, string source, bool resolve = true, Uri uri = null) {
+    public static async Task<Program> /*?*/ Parse(ErrorReporter reporter, string source, bool resolve = true, Uri uri = null, CancellationToken cancellationToken = default) {
```

Comment:
Looks like the main Dafny pipeline parse stage already has a cancellation token, so this seems obviously a good change.

Does this separate copy of some of that logic need to exist though? Should we refactor it away soon?